### PR TITLE
refactor(components): [time-select] switch to script-setup syntax

### PR DESCRIPTION
**Pull request by metanas**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/time-select/src/time-select.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  },
+  minTime: {
+    type: String,
+  },
+  maxTime: {
+    type: String,
+  },
+  name: {
+    type: String,
+  },
```
Comment (JeremyWuuuuu) [inline packages/components/time-select/src/time-select.ts:57]:
```suggestion
  minTime: String,
  maxTime: String,
  name: String,
```

File: packages/components/time-select/src/time-select.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    type: Boolean,
+    default: true,
+  },
+  size: {
+    type: String as PropType<ComponentSize>,
+    values: componentSizes,
+  },
+  placeholder: {
+    type: String,
+  },
```
Comment (JeremyWuuuuu) [inline packages/components/time-select/src/time-select.ts:36]:
```suggestion
  placeholder: String,
```

File: packages/components/time-select/src/time-select.vue [hunk 3]
```diff
@@ -30,204 +30,90 @@
...
+import { computed, ref } from 'vue'
 import dayjs from 'dayjs'
-import customParseFormat from 'dayjs/plugin/customParseFormat.js'
+import customParseFormat from 'dayjs/plugin/customParseFormat'
 import ElSelect from '@element-plus/components/select'
 import ElIcon from '@element-plus/components/icon'
-import { CircleClose, Clock } from '@element-plus/icons-vue'
 import { useNamespace } from '@element-plus/hooks'
+import { timeSelectProps } from './time-select'
+import { compareTime, formatTime, nextTime, parseTime } from './utils'
```
Comment (JeremyWuuuuu) [inline packages/components/time-select/src/time-select.vue:41]: **Praise:** this is good practice! 👍

File: packages/components/time-select/src/time-select.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+export const timeSelectProps = buildProps({
+  format: {
+    type: String,
+    default: 'HH:mm',
+  },
+  modelValue: String,
+  disabled: {
+    type: Boolean,
+    default: false,
+  },
```
Comment (JeremyWuuuuu) [inline packages/components/time-select/src/time-select.ts:17]:
```suggestion
  disabled: Boolean,
```

**Review by jw-foss — COMMENTED** [review]
I've left a few comments for you to consider 🏓

**Review by jw-foss — APPROVED** [review]

**Comment by JeremyWuuuuu** [comment]
GJ :rocket:

**Pull request merged**